lib/IRGen/GenStruct.cpp

```diff
@@ -1480,9 +1480,11 @@ class ClangRecordLowering {
     }
 
     // Otherwise, add it as an opaque blob.
-    auto fieldSize = isZeroSized ? clang::CharUnits::Zero() : ClangContext.getTypeSizeInChars(clangField->getType());
-    return addOpaqueField(offset,
-                          dataSize.value_or(Size(fieldSize.getQuantity())));
+    auto fieldTypeSize = ClangContext.getTypeSizeInChars(clangField->getType());
+    auto fieldSize = isZeroSized
+                         ? clang::CharUnits::Zero()
+                         : dataSize.value_or(Size(fieldTypeSize.getQuantity()));
```

Comment:
What are the types involved here? Do we need to wrap `fieldTypeSize.getQuantity()` in `Size`?

Reply:
Ah, I see what you mean. Yeah, let me simplify the types here.

Reply:
`fieldSize` is `clang::CharUnits`, which `swift::Size` implicitly converts to.